cpp/src/io/orc/stripe_data.cu

```diff
@@ -640,9 +783,14 @@ static __device__ uint32_t Integer_RLEv2(orc_bytestream_s* bs,
                                          T* vals,
                                          uint32_t maxvals,
                                          int t,
-                                         bool has_buffered_values = false)
+                                         bool has_buffered_values                  = false,
+                                         run_cache_manager* run_cache_manager_inst = nullptr,
```

Comment:
I haven't studied the function `gpuDecodeNullsAndStringDictionaries` yet, and I'm on the conservative side applying the cache to existing code.

I'm not quite sure about how to use `cuda::std::optional` and what's its advantage when it comes to function default arguments. Since the function will have side effect on the optional object, should I pass it by reference `cuda::std::optional<X>&`? But this way I can't give it a default value `cuda::std::nullopt`, and have to pass  `cuda::std::nullopt` to all function calls that don't need `X`. Is my understanding correct?

Reply:
Sorry, never mind. Ignore this suggestion. I forgot that you can't have optional references, which is what I was thinking of when I wrote this (`cuda::std::optional<run_cache_manager&>`).